### PR TITLE
[RfR] License Picker tweaks and fixes

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose

License picker needed a few adjustments/fixes

Also includes UI updates from: https://github.com/CenterForOpenScience/osf.io/pull/4377/

Associated tickets:
https://openscience.atlassian.net/browse/OSF-4068
https://openscience.atlassian.net/browse/OSF-4477

# Changes

- add Jeff's groupings/ordering
- fix search bugs
  - fix incorrect counts when filtering by license
  - fix issue with search bar not showing (JS error)

From 4377:
- Repositioned the Save Button
- Added a scrollbar to display the License information in a limited text area
- Changed the Cross button to make it look same as other cross buttons
- Made changes so that non-contributors/non-admins would not be able to click on the License option to choose another license
- Made Save button for choosing License visible only to admins and made it hidden for non-admins

# Side Effects 

None

### CODE REVIEW

File: website/static/js/licenses.js [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
+list.push(OTHER_LICENSE);
+
+var licenseGroups = [
+    DEFAULT_LICENSE,
+    {
+        name: 'Content:',
+        licenses: [licenses.CC0, licenses.CCBY]
+    },
+    {
+        name: 'Code = Permissive:',
```
Comment (sloria) [inline website/static/js/licenses.js]: "=" should be "-"

**Comment by sloria** [comment]
The close button still looks a little different; looks like the wrong icon

Treebeard modal:

osf___license_picker_testing

License picker:

osf___license_picker_testing

**Comment by sloria** [comment]
Just a minor change, and this is good to merge.

**Comment by sloria** [comment]
This is good to merge as is. @arpitar Can you please address the issue above with the close icon in a separate PR?

**Comment by arpitar** [comment]
Hello, thank you for the feedback. I'm working at fixing the close button.

**Pull request merged**